Tools/clinic/clinic.py

```diff
@@ -2259,7 +2264,7 @@ def parse_file(filename, *, verify=True, output=None):
             write_file(fn, data)
 
 
-def compute_checksum(input, length=None):
+def compute_checksum(input: str, length: int | None = None) -> str:
     input = input or ''
```

Comment:
This line implies to me that maybe `input` could also be `None`. If `input` can only be a `str`, then it won't be falsey unless it's the empty string, which would make this line redundant. But if `input` can be `str | None`, then this line ensures that `input` will always be a `str` for the rest of the function (so this line serves a purpose).

Reply:
Indeed it does. Good catch.